base/c.jl

```diff
@@ -23,7 +23,7 @@ if is_windows()
 else
     typealias Clong Int
     typealias Culong UInt
-    typealias Cwchar_t Int32
+    typealias Cwchar_t UInt32
```

Comment:
Also, changing this definition will be subtly breaking. Functions that need to use `Cwchar_t` should just be fixed to handle both cases.